hyperspy/signal.py

```diff
@@ -5316,6 +5316,32 @@ def map(
             lazy_output = self._lazy
         if ragged is None:
             ragged = self.ragged
+        if isinstance(silence_warnings, str):
+            silence_warnings = (silence_warnings,)
+
+        if isinstance(silence_warnings, (list, tuple)):
+            for key in silence_warnings:
+                if isinstance(key, str):
+                    if key not in ["non-uniform", "units", "scales"]:
+                        raise ValueError(
+                            f"'{key}' is not a valid string for the "
+                            "`silence_warnings` parameter."
+                        )
+                else:
+                    raise TypeError(
+                        "`silence_warnings` must be a boolean or a list, tuple of string."
+                    )
+
+        def _check_warnings(silence_warnings, key):
+            # Warn only when the key is not in `silence_warnings`
+            # or when `silence_warnings` is True
```

Comment:
Warn when `silence_warnings=True` seems counterintuitive. Does not warn if the parameter is True.